ajax-redux-agenda/src/components/Authentication.js

```diff
@@ -0,0 +1,55 @@
+import react from "react";
+
+class Authentication extends React.Component {
+  constructor() {
+    super();
+    this.state = {
+      username: "",
+      password: ""
+    };
+  }
+
+  getData = e => {
+    this.setState({
+      ...this.state,
+      [e.target.name]: e.target.value
+    });
+  };
+
+  login = e => {
+    e.preventDefault();
+    this.props.login(this.state);
+  };
```

Comment:
mai bine nu pui direct state-ul in functie, poti face mai inainte un obiect `const user={username,password}` iei din state si pui `this.props.login(user)`